src/pages/workspace/reimburse/WorkspaceReimburseView.js

```diff
@@ -167,28 +175,34 @@ class WorkspaceReimburseView extends React.Component {
                     </View>
-                    <View style={[styles.flexRow, styles.alignItemsCenter]}>
-                        <View style={[styles.rateCol]}>
-                            <TextInput
-                                label={this.props.translate('workspace.reimburse.trackDistanceRate')}
-                                placeholder={this.state.outputCurrency}
-                                onChangeText={value => this.setRate(value)}
-                                value={this.state.rateValue}
-                                autoCompleteType="off"
-                                autoCorrect={false}
-                                keyboardType={CONST.KEYBOARD_TYPE.DECIMAL_PAD}
-                                onKeyPress={this.debounceUpdateOnCursorMove}
-                            />
-                        </View>
-                        <View style={[styles.unitCol]}>
-                            <Picker
-                                label={this.props.translate('workspace.reimburse.trackDistanceUnit')}
-                                items={this.unitItems}
-                                value={this.state.unitValue}
-                                onInputChange={value => this.setUnit(value)}
-                            />
+                    <OfflineWithFeedback
+                        errors={lodashGet(this.props, ['policy', 'customUnits', this.state.unitID, 'errors'])}
+                        pendingAction={lodashGet(this.props, ['policy', 'customUnits', this.state.unitID, 'pendingAction'])}
+                        onClose={() => Policy.removeUnitError(this.props.policyID, this.state.unitID)}
+                    >
+                        <View style={[styles.flexRow, styles.alignItemsCenter, styles.mv2]}>
+                            <View style={[styles.rateCol]}>
+                                <TextInput
+                                    label={this.props.translate('workspace.reimburse.trackDistanceRate')}
+                                    placeholder={this.state.outputCurrency}
+                                    onChangeText={value => this.setRate(value)}
+                                    value={this.state.rateValue}
+                                    autoCompleteType="off"
+                                    autoCorrect={false}
+                                    keyboardType={CONST.KEYBOARD_TYPE.DECIMAL_PAD}
+                                    onKeyPress={this.debounceUpdateOnCursorMove}
+                                />
+                            </View>
+                            <View style={[styles.unitCol]}>
+                                <Picker
+                                    label={this.props.translate('workspace.reimburse.trackDistanceUnit')}
+                                    items={this.unitItems}
+                                    value={this.state.unitValue}
```

Comment:
Talking about point 3 from your [response](https://github.com/Expensify/App/pull/10374#issuecomment-1225845919)
> There is problem with WorkspaceReimburseView render.

Its still not resolved. I believe its because you have a state variable here. So when the value is updated in props the constructor is not rerun to update this unitValue. 
Check out this video to see if it helps,
I have newDot open in two windows, i update the custom unit value in one window and while its onyx is updating in the second window its UI is not.

https://user-images.githubusercontent.com/14356145/186491217-1d1e0bea-8f38-4aab-9bdb-fff6b280a54a.mov

I think you might need to use `getDerivedStateFromProps` for your usecase where you update this state value only if lodashGet(distanceCustomUnit, 'attributes.unit') has changed.

Reply:
Okay, got it for real for real this time. Thanks for explaining that further and pointing me in the right direction.

https://user-images.githubusercontent.com/7277067/186658113-9e4f330e-6c6e-4653-9e55-9836ac17f17c.mov

Reply:
Nice ✨